azure/src/main/scala/quasar/blobstore/azure/AzureListService.scala

```diff
@@ -17,31 +17,33 @@
 package quasar.blobstore.azure
 
 import quasar.blobstore.azure.requests.ListBlobHierarchyArgs
-import quasar.blobstore.paths.PrefixPath
+import quasar.blobstore.paths.{BlobstorePath, PrefixPath}
 import quasar.blobstore.services.ListService
 
 import java.lang.Integer
 
 import scala.{None, Some}
 
 import cats.data.Kleisli
-import cats.effect.{Async, ContextShift}
-import com.microsoft.azure.storage.blob.{ContainerURL, ListBlobsOptions}
-import com.microsoft.rest.v2.Context
+import cats.effect.{ConcurrentEffect, ContextShift}
+import com.azure.storage.blob.BlobContainerAsyncClient
+import com.azure.storage.blob.models.ListBlobsOptions
 
 object AzureListService {
-  def apply[F[_]: Async: ContextShift](
+
+  def apply[F[_]: ConcurrentEffect: ContextShift](
       toListBlobsOption: Kleisli[F, PrefixPath, ListBlobsOptions],
       mkArgs: ListBlobsOptions => ListBlobHierarchyArgs)
       : ListService[F] =
     toListBlobsOption map
       mkArgs andThen
       requests.listRequestK andThen
-      converters.toBlobstorePathsK
+      converters.toBlobstorePathsK andThen
+      handlers.emptyStreamToNoneK[F, BlobstorePath]
 
 
-  def mk[F[_]: Async: ContextShift](containerURL: ContainerURL): ListService[F] =
+  def mk[F[_]: ConcurrentEffect: ContextShift](containerClient: BlobContainerAsyncClient): ListService[F] =
```

Comment:
This is a massive increase in the constraint. Do we really, *really* need `ConcurrentEffect`?

Reply:
This is because the old `requests.listRequest` returned a `Single/Mono` whereas the new one returns a `Flowable/Flux`. The `ConcurrentEffect ` constraint ultimately comes from the fs2 - reactivestreams interop here https://github.com/precog/async-blobstore/pull/96/files#diff-757b40155943bc9a1eb1648c70324019R36